### PR TITLE
More elegance cleanup

### PR DESCRIPTION
**Pull request by ronawho**
- Cleanup null plan code
  - Fix missing prototype warning in C code, move extern declaration to callsite.
- Cleanup locking code
  - Move it into a helper module with links to issues requesting a locks library and using-facing locale-private vars.
 - Cleanup execute code by calling into overloads

### CODE REVIEW

**Comment by ronawho** [comment]
Misc questions for other potential cleanups
 - I don't think `isValid` is used. Could maybe be removed, but seemed worth keeping (or moved into fftw module?)
 - I _think_ setup1DPlan could just call into setupPlanColumns
 - I wondered if the cleanup called in the module deinit should be in the fftw module

**Comment by npadmana** [comment]
@ronawho  -- 
> 
> * I don't think `isValid` is used. Could maybe be removed, but seemed worth keeping (or moved into fftw module?)

Yes, I thought about deleting in in my last cleanup. And then decided against it, because I thought it might be useful to have. I agree, it should ideally be in the FFTW module as a test. Also, it would be nice to avoid the separate C code for this -- it would be nice to access NULL using the C interop features. 

Let's leave this in for now, but maybe we should open Chapel issues for these.

> * I _think_ setup1DPlan could just call into setupPlanColumns

I'm not quite sure. For the strided version, yes -- but for the unstrided version, I'm not sure. I'd be in favor of shelving this one for now.

> * I wondered if the cleanup called in the module deinit should be in the fftw module

Yes, it should (and in fact, it's in an issue chapel-lang/chapel#5751). Ditto with some of the record plan stuff. However, I wanted to defer doing those to a later date when we thought a little bit more about exactly how we should set that up.

File: src/DistributedFFT.chpl [hunk 1]
```diff
@@ -41,58 +35,46 @@ prototype module DistributedFFT {
   record FFTWplan {
     param ftType : FFTtype;
     var plan : fftw_plan;
-    var tt : TimeTracker;
 
     // Mimic the advanced interface 
-    proc init(param ftType1 : FFTtype, args ...?k) {
-      ftType = ftType1;
+    proc init(param ftType : FFTtype, args ...?k) {
+      this.ftType = ftType;
```
Comment (npadmana) [inline src/DistributedFFT.chpl]: Aha! That's the syntax I should remember.

**Comment by npadmana** [comment]
These look good. I think after we get through our list of merges, we should probably rerun the perf suite again, just to make sure we haven't regressed somewhere.

I'm ready to merge this, if you are.

**Comment by ronawho** [comment]
> These look good. I think after we get through our list of merges, we should probably rerun the perf suite again, just to make sure we haven't regressed somewhere.

Yeah, I ran size D/E up to 256 nodes, but was planning on doing full runs once we were done.

> I'm ready to merge this, if you are.

Yup

**Comment by ronawho** [comment]
I have one last potential change, and I should be done after that. Working on the PR now.

**Pull request merged**